datasetinsights/storage/checkpoint.py

```diff
@@ -5,89 +5,109 @@
+                "supported".format(path)
+            )
 
+        return method
 
-def create_checkpointer(*, logdir, config):
-    """
-    Initialize the correct estimator checkpointer
-    Args:
-        logdir: filepath to where to save/load local copy of estimator
-        config:
+    def save(self, estimator, epoch):
+        """
+        Saves estimator to log directory
 
-    Returns (EstimatorCheckpoint): the correct estimator checkpoint for the
-    config. The Estimator Checkpoint is responsible for saving and loading
-    the estimator.
+        Args:
+            estimator (datasetinsights.estimators.Estimator):
+            datasetinsights estimator object
+            epoch (int): epoch number
+        """
+        if self.distributed and not is_master():
+            return
+        self._writer.save(estimator=estimator, epoch=epoch)
 
-    """
-    if logdir.startswith(GCS_BASE_STR):
-        checkpointer = GCSEstimatorCheckpoint(logdir, config.estimator)
-    elif checkpoint_file_on_gcs(config):
-        logdir = f"{GCS_BASE_STR}{const.GCS_BUCKET}/runs/{str(logdir)}"
-        checkpointer = GCSEstimatorCheckpoint(logdir, config.estimator)
-    else:
-        checkpointer = EstimatorCheckpoint(logdir, config.estimator)
-    if config.system.distributed:
-        checkpointer = DistributedEstimatorCheckpoint(
-            is_master=is_master(), estimator_checkpoint=checkpointer
-        )
-    return checkpointer
+    def load(self, estimator, path):
+        """ Loads estimator from given path
 
+        Path can be either a local path or GCS path or HTTP url
 
-class EstimatorCheckpoint:
-    """ Interact with estimator checkpoints
+        Args:
+            estimator (datasetinsights.estimators.Estimator):
+            datasetinsights estimator object
+            path (str): path of estimator
+        """
+        if self.distributed and not is_master():
+            return
+
+        load_method = self._get_loader_from_path(path)
+        load_method(estimator, path)
+
+
+class LocalEstimatorWriter:
+    """ Writes (saves) estimator checkpoints locally
 
     Args:
-        dirname (str): directory name of where checkpoint files are stored
         prefix (str): filename prefix of the checkpoint files
         suffix (str): filename suffix of the checkpoint files
-        create_dir (bool): indicate whether to force create directoy if
-            the specified dirname does not exist. Default: True
 
     Attributes:
         dirname (str): directory name of where checkpoint files are stored
         prefix (str): filename prefix of the checkpoint files
         suffix (str): filename suffix of the checkpoint files
     """
 
-    def __init__(
-        self, dirname, prefix, *, suffix=DEFAULT_SUFFIX, create_dir=True
-    ):
-        self.dirname = dirname
+    def __init__(self, prefix, *, suffix=DEFAULT_SUFFIX):
+        self.dirname = os.path.join(const.PROJECT_ROOT, prefix)
```

Comment:
Why do we have to remove the original and assumes estimator will create under `PROJECT_ROOT`? The original design allow user to use any estimator dir to any location.

Reply:
Fixed this.